pylint/pyreverse/diadefslib.py

```diff
@@ -21,29 +23,30 @@
 class DiaDefGenerator:
     """Handle diagram generation options."""
 
-    def __init__(self, linker, handler):
+    def __init__(self, linker: Linker, handler: DiadefsHandler) -> None:
         """Common Diagram Handler initialization."""
         self.config = handler.config
+        self.module_names: bool = False
         self._set_default_options()
         self.linker = linker
-        self.classdiagram = None  # defined by subclasses
+        self.classdiagram: ClassDiagram | None = None  # defined by subclasses
```

Comment:
In some instances of this pattern in `pylint` we have chosen to do:
```suggestion
        self.classdiagram: ClassDiagram  # defined by subclasses
```

To avoid having to do `if self.classdiagram` everywhere. Is that something we should consider here too?

Reply:
A pattern I like is to have for something like this:
```python
from abc import ABC, abstractmethod

class Animal(ABC):

    @classmethod
    @property
    @abstractmethod
    def sound_maker_class(cls):
        raise NotImplementedError
    
    def __init__(self):
        self.sound_maker = sound_maker_class()

class Cat:
    sound_maker_class = Meower
```